src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -687,12 +728,32 @@ private static bool IsArrayCompatibleReadOnlyInterface(Type type)
             return allProperties;
         }
 
+        [RequiresUnreferencedCode(PropertyTrimmingWarningMessage)]
+        private static object? BindParameter(ParameterInfo property, IConfiguration config, BinderOptions options)
+        {
+            string parameterName = property.Name!;
+
+            var propertyBindingPoint = new BindingPoint(initialValue: config.GetSection(parameterName).Value, isReadOnly: false);
```

Comment:
We shouldn't stringify an entire section. The initial value should be the parameters default value if anything.

```suggestion
            var initialValue = parameter.HasDefaultValue ? parameter.DefaultValue : null;
            var propertyBindingPoint = new BindingPoint(initialValue);
```

We should have test cases for this.